app/ui.js

```diff
@@ -435,6 +440,16 @@ var UI;
                     document.documentElement.classList.add("noVNC_disconnecting");
                     break;
                 case 'disconnected':
+                    var reconnect = WebUtil.getConfigVar('reconnect');
+
+                    if ((reconnect === 'true' || reconnect == '1') && UI.reconnect_enabled) {
+                        UI.showStatus(_("Reconnecting"));
+
+                        var delay = parseInt(WebUtil.getConfigVar('reconnect_delay', UI.reconnect_delay));
+                        setTimeout(UI.connect, delay);
```

Comment:
A good start, but two issues:

 a) How do you abort the reconnect, if that wasn't what you wanted?
 b) We should block access to the rest of the UI whilst we're waiting for this timer or the user could easily get things in some odd state.